main/realmain.cc

```diff
@@ -270,27 +271,30 @@ void runAutogen(const core::GlobalState &gs, options::Options &opts, const autog
 
     autogen::DefTree root;
     AutogenResult out;
-    vector<pair<int, AutogenResult::Serialized>> merged;
-    for (auto res = resultq->wait_pop_timed(out, chrono::seconds{1}, *logger); !res.done();
-         res = resultq->wait_pop_timed(out, chrono::seconds{1}, *logger)) {
+    for (auto res = resultq->wait_pop_timed(out, WorkerPool::BLOCK_INTERVAL(), *logger); !res.done();
+         res = resultq->wait_pop_timed(out, WorkerPool::BLOCK_INTERVAL(), *logger)) {
         if (!res.gotItem()) {
             continue;
         }
         counterConsume(move(out.counters));
-        merged.insert(merged.end(), make_move_iterator(out.prints.begin()), make_move_iterator(out.prints.end()));
+        for (auto &print : out.prints) {
+            merged[print.first] = move(print.second);
+        }
         if (opts.print.AutogenAutoloader.enabled) {
             Timer timeit(logger, "autogenAutoloaderDefTreeMerge");
             root = autogen::DefTreeBuilder::merge(gs, move(root), move(*out.defTree));
         }
     }
-    fast_sort(merged, [](const auto &lhs, const auto &rhs) -> bool { return lhs.first < rhs.first; });
```

Comment:
Are you sure we can drop this sort? Do we really not need it for the purposes of making our exp files consistent?

Reply:
@jez Read the PR. I am not changing sort order. :) The output comes bundled with its final index!

Reply:
Output is a pair of <index, output data>. Rather than make a vector of pairs of indices and output data and sorting by indices, I create a presized vector of output data and slot in the output into the appropriate index.

Reply:
Makes sense, thanks!

Reply:
OOC: Did you consider handing the thread a pointer into `merged` to write to instead of writing out the idx's? Or is that worse for thread-pinning issues?

Reply:
I think that would be worse as it would cause cache contention on the `merged` vector. Core 1 would have to load part of the vector exclusively into memory, write the output into it, and flush it to main memory before core 2 could load it up exclusively to do the same.